code/modules/surgery/organs/wings.dm

```diff
@@ -145,9 +145,63 @@
 	name = "pair of bee wings"
 	desc = "A pair of bee wings. They seem tiny and undergrown"
 	icon_state = "beewings"
-	flight_level = WINGS_FLIGHTLESS
+	flight_level = WINGS_COSMETIC
+	actions_types = list(/datum/action/item_action/organ_action/use/bee_dash)
 	wing_type = "Bee"
 
+/datum/action/item_action/organ_action/use/bee_dash
+	var/jumpdistance = 3
+	var/jumpspeed = 1
+	var/recharging_rate = 100
+	var/recharging_time = 0
+
+/datum/action/item_action/organ_action/use/bee_dash/Trigger()
+	var/mob/living/carbon/L = owner
+
+	if(L.stat != CONSCIOUS || L.buckling) // Has to be concious and unbuckled
+		return
+	if(recharging_time > world.time)
+		to_chat(L, "<span class='warning'>The wings aren't ready to dash yet!</span>")
+		return
+	var/datum/gas_mixture/environment = L.loc.return_air()
+	if(environment && !(environment.return_pressure() > 30))
+		to_chat(L, "<span class='warning'>The atmosphere is too thin for you to dash!</span>")
+		return
+
+	var/atom/target = get_edge_target_turf(L, L.dir) //gets the user's direction
+	var/hoppingtable = FALSE // Triggers the trip
+	var/jumpdistancemoved = jumpdistance // temp jumpdistance
+	var/turf/checkjump = get_turf(L)
+	for(var/i in 1 to jumpdistance) //This is how hiero club find the tiles in front of it, tell me/fix it if there's a better way
+		if(locate(/obj/structure/table, get_step(checkjump, L.dir))) // If there's a table, trip
+			hoppingtable = TRUE
+			jumpdistancemoved = i
+			break
+		var/turf/T = get_turf(get_step(checkjump, L.dir))
+		if(T.density == TRUE) // If turf has density (wall), break
+			break
+		var/dense_in_the_way = FALSE
+		for(var/atom/movable/A in T) // breaks if there's anything solid in the way, no tripping on tables through walls
+			if(A.density == TRUE)
+				dense_in_the_way = TRUE
+				break
+		if(dense_in_the_way == TRUE)
+			break
```

Comment:
```suggestion
		for(var/atom/movable/A in T) // breaks if there's anything solid in the way, no tripping on tables through walls
			if(A.density)
				break
```

Reply:
This only breaks `for(var/atom/movable/A in T)`, and not the whole loop

Reply:
You want to break the whole loop when there is something "dense in the way", eh? My suggested edit achieves the same result.

Reply:
You can always just `break` to a label. I'm not sure if there's any reason not to use labels.

Reply:
I used labels, thanks ike